overlord/snapstate/check_snap.go

```diff
@@ -613,10 +622,20 @@ var osutilEnsureUserGroup = osutil.EnsureUserGroup
 
 func validateSystemUsernames(si *snap.Info) error {
 	for _, user := range si.SystemUsernames {
-		if _, ok := supportedSystemUsernames[user.Name]; !ok {
+		systemUserName, ok := supportedSystemUsernames[user.Name]
+		if !ok {
 			return fmt.Errorf(`snap %q requires unsupported system username "%s"`, si.InstanceName(), user.Name)
 		}
 
+		if systemUserName.AllowedSnapIds != nil {
```

Comment:
this check then should look use si.SnapID,  and here we should check whether  `&& si.SnapID != ""`